File: C/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int getChangeOwned(int value);
int nextCoin(int coin);

int main(void)
{
    int changeOwned;

    do
    {
        changeOwned = get_int("Change Owned: ");
    }while(changeOwned<0);

    changeOwned = getChangeOwned(changeOwned);

    printf("%i\n",changeOwned);
}

int getChangeOwned(int value)
{
    int changeOwned = 0;
    int coinValue   = 25;

    if(value<0)
        return changeOwned;

    while(value>0){
        if(coinValue>value){
            coinValue = nextCoin(coinValue);
            continue;
        }

        value-=coinValue;
        changeOwned++;
    }

    return changeOwned;
}

int nextCoin(int coin)
{
    if(coin==25)
        return 10;
    else if(coin==10)
        return 5;

    return 1;
}

File: Python/sentimental-cash/cash.py
from cs50 import get_float

while True:
    change = get_float("Change: ")
    if(change > 0):
        break

changeOwned = 0
coins = [0.25, 0.10, 0.05, 0.01]
coin = coins[0]
i = 0

while change>0:
    if coin > change:
        i += 1
        coin = coins[3] if (i > 3) else coins[i]
        continue

    change = round(change - coin, 2)
    changeOwned+=1

print(changeOwned)
